src/Extensions/Startup/OmexApplicationStartup.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.Omex.Extensions.Compatability;
+using Microsoft.Omex.Extensions.Logging;
+
+namespace Microsoft.Omex.Extensions.ServiceFabric
+{
+	/// <summary>
+	/// Class to start application with initialized DI and Logging
+	/// </summary>
+	/// <typeparam name="TContext">Service context implentation type used for logging, use NullServiceContext if it's not required</typeparam>
+	public class OmexApplicationStartup<TContext>
+		where TContext : class, IServiceContext
+	{
+		/// <summary>Create an instance of OmexApplicationStartup</summary>
+		/// <param name="serviceTypeName">Name of the service used for logging</param>
+		public OmexApplicationStartup(string? serviceTypeName = null) =>
+			ServiceTypeName = serviceTypeName ?? GetDefaultName();
+
+
+		/// <summary>Run function to execute</summary>
+		/// <param name="function">Function to execute</param>
+		/// <param name="typeRegestration">Action to register types in DI</param>
+		/// <param name="typeResolution">Action to resolve types from DI</param>
+		/// <returns></returns>
+		protected async Task RunAsync(
+			Func<Task> function,
+			Action<IServiceCollection>? typeRegestration = null,
+			Action<IServiceProvider>? typeResolution = null)
+		{
+			try
+			{
+				m_typeRegestration = typeRegestration;
+				m_typeResolution = typeResolution;
+				Task task = function();
+				LogSuccess(Process.GetCurrentProcess().Id, ServiceTypeName);
+				await task;
+			}
+			catch (Exception exception)
+			{
+				LogFailure(exception);
+				throw;
+			}
+		}
+
+
+		/// <summary>Register dependencies in DI</summary>
+		protected virtual IServiceCollection Register(IServiceCollection collection)
+		{
+			collection
+				.AddOmexLogging<TContext>()
+				.AddTimedScopes();
+			m_typeRegestration?.Invoke(collection);
+			return collection;
+		}
+
+
+		/// <summary>Resolve dependencies from DI</summary>
+		protected virtual IServiceProvider Resolve(IServiceProvider provider)
+		{
+			provider.InitializeOmexCompatabilityClasses();
+			m_typeResolution?.Invoke(provider);
+			return provider;
+		}
+
+
+		/// <summary></summary>
+		/// <typeparam name="T">Type of startup objest </typeparam>
```

Comment:
>T [](start = 23, length = 1)

Please use a more descriptive type than `T`

Reply:
Code deleted, but I'll take it into account